repository_pool.go

```diff
@@ -178,17 +188,17 @@ func (p *RepositoryPool) AddGit(path string) error {
 
 // AddGitWithID adds a git repository to the pool. ID should be specified.
 func (p *RepositoryPool) AddGitWithID(id, path string) error {
-	return p.Add(gitRepo(id, path))
+	return p.Add(gitRepo(id, path, p.cache))
```

Comment:
I wouldn't store the cache in the git/sivaRepo. If they are different instances (one per repo) it can use lots of memory.